### DISCUSSION TITLE
How to configure default dynamic route value?

### DISCUSSION THREAD

**Discussion by kalkusa** [vercel/next.js#26449] [Help]
The idea behind dynamic routes is simple and therefore the documentation is rather limited. I'm working on the app which uses two dynamic route parameters - `theme` and `locale`. I would like to configure default value for the latter parameter, so it could be omitted by the user.

For example instead of `/foo/blue/en` the user can navigate to `/foo/blue`

I haven't been able to find such option in the documentation. Is it possible to achieve in Next.js?

Alternatively - can I intercept redirection to 404 in such case if default dynamic route values aren't implemented (so I can redirect to default route)?

**Comment by tinywinds** [comment]
This problem can be solved with nginx's rewrite. The default value is contrary to not-found.js.

**Comment by hwsdien** [comment]
If you use `[slug]` or `[...slug]` in Next.js without providing a `slug`, the route may not render. To address this, you can change `[slug]` to `[[...slug]]`, which ensures that the route matches even when `slug` is not provided. In the code, you can check for the presence of `slug` using `!params.slug ? default_value : params.slug[0]`. It's important to note that this method will capture all subsequent route segments, so you can handle the logic accordingly by checking if the `params.slug` array length is equal to 1.
